Data/models/Bookmark.swift

```diff
@@ -319,104 +265,85 @@ public class Bookmark: NSManagedObject, WebsitePresentable, Syncable {
         // If I save while the animation is happening, the rows look screwed up (draw on top of each other).
         // Adding a delay to let animation complete avoids this problem
         DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(250)) {
-            DataController.saveContext(context: frc?.managedObjectContext)
+            DataController.save(frc?.managedObjectContext)
         }
 
     }
 }
 
 // TODO: Document well
-// Getters
+// MARK: - Getters
 extension Bookmark {
-    fileprivate static func get(forUrl url: URL, countOnly: Bool = false, getFavorites: Bool = false, context: NSManagedObjectContext) -> AnyObject? {
-        let fetchRequest = NSFetchRequest<NSFetchRequestResult>()
-        fetchRequest.entity = Bookmark.entity(context: context)
+    fileprivate static func get(forUrl url: URL, countOnly: Bool = false, getFavorites: Bool = false, context: NSManagedObjectContext) -> [Bookmark]? {
+        let urlKeyPath = #keyPath(Bookmark.url)
+        let isFavoriteKeyPath = #keyPath(Bookmark.isFavorite)
+        
         let isFavoritePredicate = getFavorites ? "YES" : "NO"
-        fetchRequest.predicate = NSPredicate(format: "url == %@ AND isFavorite == \(isFavoritePredicate)", url.absoluteString)
-        do {
-            if countOnly {
-                let count = try context.count(for: fetchRequest)
-                return count as AnyObject
-            }
-            let results = try context.fetch(fetchRequest) as? [Bookmark]
-            return results?.first
-        } catch {
-            let fetchError = error as NSError
-            print(fetchError)
-        }
-        return nil
+        let predicate = NSPredicate(format: "\(urlKeyPath) == %@ AND \(isFavoriteKeyPath) == \(isFavoritePredicate)", url.absoluteString)
+        
+        return all(where: predicate)
     }
     
-    public static func getChildren(forFolderUUID syncUUID: [Int]?, ignoreFolders: Bool = false, context: NSManagedObjectContext,
-                            orderSort: Bool = false) -> [Bookmark]? {
+    fileprivate static func count(forUrl url: URL, getFavorites: Bool = false) -> Int? {
```

Comment:
Besides the last call, is this function equivalent to `get(forURL`? If so, we should abstract better.